Stardew not being detected

# Missing Game Report

For some reason it is unable to find stardew valley on my system. It is being run through the Heroic Launcher but is located on a different drive than the main system, I do not know if this would effect anything but this might be a problem. I have tried to re-install it a few times but does not make a difference and is still not detected.

For context if needed the game is located at a custom mount point at /mnt/stuff/Game/Heroic/Stardew Valley.

## Name

Stardew Valley

## Store

GOG

## Operating System

Fedora Linux

## Attach Logs

[nexusmods.app.main.current.log](https://github.com/user-attachments/files/20302115/nexusmods.app.main.current.log)
[nexusmods.app.slim.current.log](https://github.com/user-attachments/files/20302116/nexusmods.app.slim.current.log)

```
00:00:00.815 [ERROR] (NexusMods.DataModel.GameRegistry.GameRegistry) Exception in locator|System.Text.Json.JsonException: The JSON value could not be converted to GameFinder.Launcher.Heroic.DTOs.Installed. Path: $.installed[2].buildId | LineNumber: 38 | BytePositionInLine: 22.
 ---> System.InvalidOperationException: Cannot get the value of a token type 'Number' as a string.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_ExpectedString(JsonTokenType tokenType)
   at System.Text.Json.Utf8JsonReader.GetString()
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)
   at System.Text.Json.Serialization.JsonConverter`1.TryReadAsObject(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, Object& value)
   at System.Text.Json.Serialization.Converters.LargeObjectWithParameterizedConstructorConverter`1.ReadAndCacheConstructorArgument(ReadStack& state, Utf8JsonReader& reader, JsonParameterInfo jsonParameterInfo)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.ReadConstructorArgumentsWithContinuation(ReadStack& state, Utf8JsonReader& reader, JsonSerializerOptions options)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)
   at System.Text.Json.Serialization.JsonCollectionConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TCollection& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)
   at System.Text.Json.Serialization.Converters.SmallObjectWithParameterizedConstructorConverter`5.TryRead[TArg](ReadStack& state, Utf8JsonReader& reader, JsonParameterInfo jsonParameterInfo, TArg& arg)
   at System.Text.Json.Serialization.Converters.SmallObjectWithParameterizedConstructorConverter`5.ReadAndCacheConstructorArgument(ReadStack& state, Utf8JsonReader& reader, JsonParameterInfo jsonParameterInfo)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.ReadConstructorArgumentsWithContinuation(ReadStack& state, Utf8JsonReader& reader, JsonSerializerOptions options)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, T& value, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, Utf8JsonReader& reader, Exception ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, T& value, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo`1.ContinueDeserialize(ReadBufferState& bufferState, JsonReaderState& jsonReaderState, ReadStack& readStack, T& value)
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo`1.Deserialize(Stream utf8Json)
   at GameFinder.Launcher.Heroic.HeroicGOGHandler.ParseInstalledJsonFile(AbsolutePath path, AbsolutePath configPath)+MoveNext()
   at GameFinder.Launcher.Heroic.HeroicGOGHandler.FindAllGames()+MoveNext()
   at System.Linq.Enumerable.<ToArray>g__EnumerableToArray|314_0[TSource](IEnumerable`1 source)
   at GameFinder.Common.Extensions.SplitResults[TGame](IEnumerable`1 results)
   at GameFinder.Common.AHandler`2.FindAllGamesById(ErrorMessage[]& errors)
   at NexusMods.StandardGameLocators.HeroicGogLocator.Find(ILocatableGame game)+MoveNext() in /_/src/NexusMods.StandardGameLocators/HeroicGogLocator.cs:line 37
   at NexusMods.DataModel.GameRegistry.GameRegistry.FindInstallations()+MoveNext() in /_/src/NexusMods.DataModel/GameRegistry/GameRegistry.cs:line 157
```

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -18,14 +18,14 @@
     <PackageVersion Include="Nerdbank.Streams" Version="2.11.79" />
     <PackageVersion Include="NexusMods.Cascade" Version="0.12.0" />
     <PackageVersion Include="NexusMods.Cascade.SourceGenerator" Version="0.12.0" />
-    <PackageVersion Include="NexusMods.MnemonicDB" Version="0.13.0" />
-    <PackageVersion Include="NexusMods.MnemonicDB.Abstractions" Version="0.13.0" />
-    <PackageVersion Include="NexusMods.Hashing.xxHash3.Paths" Version="3.0.3" />
-    <PackageVersion Include="NexusMods.Hashing.xxHash3" Version="3.0.3" />
+    <PackageVersion Include="NexusMods.MnemonicDB" Version="0.13.1" />
+    <PackageVersion Include="NexusMods.MnemonicDB.Abstractions" Version="0.13.1" />
+    <PackageVersion Include="NexusMods.Hashing.xxHash3.Paths" Version="3.0.4" />
+    <PackageVersion Include="NexusMods.Hashing.xxHash3" Version="3.0.4" />
     <PackageVersion Include="NexusMods.Archives.Nx" Version="0.6.4" />
-    <PackageVersion Include="NexusMods.Paths" Version="0.19.0" />
-    <PackageVersion Include="NexusMods.Paths.Extensions.Nx" Version="0.19.0" />
-    <PackageVersion Include="NexusMods.Paths.TestingHelpers" Version="0.19.0" />
+    <PackageVersion Include="NexusMods.Paths" Version="0.19.1" />
+    <PackageVersion Include="NexusMods.Paths.Extensions.Nx" Version="0.19.1" />
+    <PackageVersion Include="NexusMods.Paths.TestingHelpers" Version="0.19.1" />
     <PackageVersion Include="NexusMods.Telemetry.OpenTelemetry" Version="1.0.0" />
     <PackageVersion Include="FomodInstaller.Interface" Version="1.2.0" />
     <PackageVersion Include="FomodInstaller.Scripting.XmlScript" Version="1.0.0" />
@@ -145,7 +145,7 @@
     <PackageVersion Include="BitFaster.Caching" Version="2.5.2" />
     <PackageVersion Include="CliWrap" Version="3.6.7" />
     <PackageVersion Include="DynamicData" Version="9.2.2" />
-    <PackageVersion Include="GameFinder" Version="4.6.3" />
+    <PackageVersion Include="GameFinder" Version="4.7.2" />
     <PackageVersion Include="Humanizer" Version="2.14.1" />
     <PackageVersion Include="ini-parser-netstandard" Version="2.5.2" />
     <PackageVersion Include="Mutagen.Bethesda.Skyrim" Version="0.44.0" />
@@ -155,6 +155,7 @@
     <PackageVersion Include="ReactiveUI.Fody" Version="19.5.41" />
     <PackageVersion Include="Sewer56.BitStream" Version="1.3.0" />
     <PackageVersion Include="Splat.Microsoft.Extensions.Logging" Version="15.2.22" />
-    <PackageVersion Include="TransparentValueObjects" Version="1.0.2" />
+    <PackageVersion Include="TransparentValueObjects" Version="1.1.0" />
+      <PackageVersion Include="TransparentValueObjects.Abstractions" Version="1.1.0" />
   </ItemGroup>
 </Project>

--- a/src/NexusMods.Abstractions.GameLocators/Stores/GOG/GOGLocatorResultMetadata.cs
+++ b/src/NexusMods.Abstractions.GameLocators/Stores/GOG/GOGLocatorResultMetadata.cs
@@ -16,13 +16,13 @@ public record GOGLocatorResultMetadata : IGameLocatorResultMetadata
     /// <summary>
     /// The build ID of the found game.
     /// </summary>
-    public required string BuildId { get; init; }
+    public required ulong BuildId { get; init; }
     
     /// <inheritdoc/>
     public ILinuxCompatibilityDataProvider? LinuxCompatibilityDataProvider { get; init; }
 
     /// <inheritdoc />
-    public IEnumerable<LocatorId> ToLocatorIds() => [LocatorId.From(BuildId)];
+    public IEnumerable<LocatorId> ToLocatorIds() => [LocatorId.From(BuildId.ToString())];
 }
 
 [PublicAPI]

--- a/src/NexusMods.Abstractions.Jobs/NexusMods.Abstractions.Jobs.csproj
+++ b/src/NexusMods.Abstractions.Jobs/NexusMods.Abstractions.Jobs.csproj
@@ -5,6 +5,7 @@
     <ItemGroup>
         <PackageReference Include="DynamicData" />
         <PackageReference Include="TransparentValueObjects" PrivateAssets="all" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+        <PackageReference Include="TransparentValueObjects.Abstractions" />
         <PackageReference Include="NexusMods.MnemonicDB.SourceGenerator" PrivateAssets="all" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
         <PackageReference Include="NexusMods.MnemonicDB.Abstractions" />
     </ItemGroup>

--- a/src/NexusMods.Abstractions.Loadouts/NexusMods.Abstractions.Loadouts.csproj
+++ b/src/NexusMods.Abstractions.Loadouts/NexusMods.Abstractions.Loadouts.csproj
@@ -12,6 +12,7 @@
       <ProjectReference Include="..\NexusMods.Abstractions.Serialization\NexusMods.Abstractions.Serialization.csproj" />
       <PackageReference Include="NexusMods.MnemonicDB.Abstractions" />
       <PackageReference Include="TransparentValueObjects" PrivateAssets="all" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+        <PackageReference Include="TransparentValueObjects.Abstractions" />
       <PackageReference Include="NexusMods.MnemonicDB.SourceGenerator" PrivateAssets="all" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
       <PackageReference Include="NexusMods.Cascade.SourceGenerator" PrivateAssets="all" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
     </ItemGroup>

--- a/src/NexusMods.Abstractions.MnemonicDB.Attributes/NexusMods.Abstractions.MnemonicDB.Attributes.csproj
+++ b/src/NexusMods.Abstractions.MnemonicDB.Attributes/NexusMods.Abstractions.MnemonicDB.Attributes.csproj
@@ -3,7 +3,9 @@
     <Import Project="$([MSBuild]::GetPathOfFileAbove('NuGet.Build.props', '$(MSBuildThisFileDirectory)../'))" />
 
     <ItemGroup>
-      <PackageReference Include="NexusMods.MnemonicDB.Abstractions" />
+        <PackageReference Include="NexusMods.MnemonicDB.Abstractions" />
+        <PackageReference Include="TransparentValueObjects" PrivateAssets="all" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+        <PackageReference Include="TransparentValueObjects.Abstractions" />
     </ItemGroup>
 
     <ItemGroup>

--- a/src/NexusMods.Abstractions.Settings/NexusMods.Abstractions.Settings.csproj
+++ b/src/NexusMods.Abstractions.Settings/NexusMods.Abstractions.Settings.csproj
@@ -11,6 +11,7 @@
         <PackageReference Include="DynamicData" />
 
         <PackageReference Include="TransparentValueObjects" PrivateAssets="all" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+        <PackageReference Include="TransparentValueObjects.Abstractions" />
     </ItemGroup>
 
     <ItemGroup>

--- a/src/NexusMods.Sdk/NexusMods.Sdk.csproj
+++ b/src/NexusMods.Sdk/NexusMods.Sdk.csproj
@@ -8,6 +8,7 @@
         <PackageReference Include="Polly.Core" />
         <PackageReference Include="R3" />
         <PackageReference Include="TransparentValueObjects" PrivateAssets="all" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+        <PackageReference Include="TransparentValueObjects.Abstractions" />
     </ItemGroup>
 
 </Project>

--- a/tests/NexusMods.StandardGameLocators.TestHelpers/Services.cs
+++ b/tests/NexusMods.StandardGameLocators.TestHelpers/Services.cs
@@ -43,7 +43,7 @@ public static class Services
     {
         coll.AddSingleton<AHandler<GOGGame, GOGGameId>>(s =>
             new StubbedGameLocator<GOGGame, GOGGameId>(s.GetRequiredService<TemporaryFileManager>(),
-                tfm => new GOGGame(GOGGameId.From(gogGameId), name, tfm.CreateFolder("gog_game").Path, buildId.ToString()),
+                tfm => new GOGGame(GOGGameId.From(gogGameId), name, tfm.CreateFolder("gog_game").Path, buildId),
                 game => game.Id));
         return coll;
     }
@@ -112,7 +112,7 @@ public static class Services
 
         coll.AddSingleton<AHandler<GOGGame, GOGGameId>>(s =>
             new StubbedGameLocator<GOGGame, GOGGameId>(s.GetRequiredService<TemporaryFileManager>(),
-                tfm => new GOGGame(GOGGameId.From(42), "Stubbed Game", tfm.CreateFolder("gog_game").Path, "4242"),
+                tfm => new GOGGame(GOGGameId.From(42), "Stubbed Game", tfm.CreateFolder("gog_game").Path, 4242),
                 game => game.Id));
 
         var steamId = "StubbedGameState.zip".xxHash3AsUtf8().Value;

